src/embed/connection.rs

```diff
@@ -461,13 +464,24 @@ async fn do_get(read: db::Read<'_>, req: GetRequest) -> Result<GetResponse, Stri
     })
 }
 
-async fn do_scan(read: db::Read<'_>, req: ScanRequest) -> Result<ScanResponse, String> {
-    use std::convert::TryFrom;
-    let mut res = Vec::<ScanItem>::new();
+async fn do_scan(
+    read: db::Read<'_>,
+    req: ScanRequest,
+    req_raw: JsValue,
+) -> Result<ScanResponse, ScanError> {
+    use ScanError::*;
+    let receiver: Function = Reflect::get(&req_raw, &JsValue::from_str("receiver"))
+        .map_err(|_| MissingReceiver)?
+        .dyn_into()
+        .map_err(|_| InvalidReceiver)?;
+
     for pe in read.scan((&req.opts).into()) {
-        res.push(ScanItem::try_from(pe).map_err(to_debug)?);
+        let key = JsValue::from_str(std::str::from_utf8(pe.key).unwrap());
+        let val = unsafe { Uint8Array::view(pe.val) };
+        // TODO: receiver can return to us whether to keep going!
```

Comment:
Can you elaborate on this comment? Like what if the receiver decides to do a scan?

Reply:
I was saying that the receiver could return a boolean to tell us whether to get the next value, and we could potentially simplify batching/limits.